apps/web/components/dashboard/tags/AllTagsView.tsx

```diff
@@ -54,48 +57,128 @@ function DeleteAllUnusedTags({ numUnusedTags }: { numUnusedTags: number }) {
     </ActionConfirmingDialog>
   );
 }
-
-function TagPill({
-  id,
-  name,
-  count,
-}: {
-  id: string;
-  name: string;
-  count: number;
-}) {
-  return (
-    <div className="group relative flex">
-      <Link
-        className="flex gap-2 rounded-md border border-border bg-background px-2 py-1 text-foreground hover:bg-foreground hover:text-background"
-        href={`/dashboard/tags/${id}`}
-      >
-        {name} <Separator orientation="vertical" /> {count}
-      </Link>
-
-      <DeleteTagConfirmationDialog tag={{ name, id }}>
-        <Button
-          size="none"
-          variant="secondary"
-          className="-translate-1/2 absolute -right-1 -top-1 hidden rounded-full group-hover:block"
-        >
-          <X className="size-3" />
-        </Button>
-      </DeleteTagConfirmationDialog>
-    </div>
-  );
+interface DragState {
+  dragId: string | null;
+  dragOverId: string | null;
+  x: number;
+  y: number;
 }
 
+const initialState: DragState = {
+  dragId: "",
+  dragOverId: "",
+  x: 0,
+  y: 0,
+};
+let currentState: DragState = initialState;
```

Comment:
I'm not a react expert but I think global mutable state are frowned upon? Is it not possible to do this with a `React.useState` instead?

Reply:
So, after learning a lot, i found this:
useState needs to be used for the change detection to work, but the drag&drop handlers are quite slow to execute and react-draggable is not implemented that well. Due to the processing overhead, the <div> will drift away from the mouse over time (lagging behind) and it looks stupid. 
My followup PR for also implementing drag&drop for assigning bookmarks to lists has therefore some changes:
* The components themselves keep their state instead of the parent --> more fine grained rerender decisions possible
* most of the state is kept outside of useState and instead in a simple closure, to make sure everything is still fast enough
* the isDragging state is kept with useState, which makes a rerender possible, when the user starts to drag.